src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java

```diff
@@ -93,13 +94,18 @@ public Payment findPaymentForExternalId(String externalId) {
     public Payment createAndCollectPayment(GatewayAccount gatewayAccount, Mandate mandate,
                                            CollectPaymentRequest collectPaymentRequest) {
 
-        Payment payment = createPayment(collectPaymentRequest, mandate, gatewayAccount.getExternalId());
+        Payment createdPayment = createPayment(collectPaymentRequest, mandate, gatewayAccount.getExternalId());
 
-        LocalDate chargeDate = paymentProviderFactory
+        var providerPaymentIdAndChargeDate = paymentProviderFactory
```

Comment:
My opinion would be the type is useful here (rather than var) because it's not obvious from the method name what this will return. Thoughts?

Reply:
It’s not obvious from the diff but the return type is no longer `LocalDate` and is instead `PaymentProviderPaymentIdAndChargeDate`, which is almost identical to the variable name, hence the use of `var`.